x-pack/plugins/enterprise_search/common/types/connectors.ts

```diff
@@ -160,15 +160,25 @@ export enum TriggerMethod {
   SCHEDULED = 'scheduled',
 }
 
+export enum SyncJobType {
+  FULL = 'full',
+  INCREMENTAL = 'incremental',
+  ACCESS_CONTROL = 'access_control',
+}
+
 export enum FeatureName {
   FILTERING_ADVANCED_CONFIG = 'filtering_advanced_config',
   FILTERING_RULES = 'filtering_rules',
+  DOCUMENT_LEVEL_SECURITY = 'document_level_security',
+  INCREMENTAL_SYNC = 'incremental_sync',
   SYNC_RULES = 'sync_rules',
 }
 
 export type ConnectorFeatures = Partial<{
+  [FeatureName.DOCUMENT_LEVEL_SECURITY]: boolean;
```

Comment:
Should these be `{ enabled: boolean }` as well ?  @timgrein

Reply:
should be, yes